src/pages/Login/Login.jsx

```diff
@@ -1,11 +1,171 @@
-import React from 'react'
-import "./Login.css"
+import { useState } from 'react'
+import { useNavigate } from "react-router-dom"
+import { 
+  ChakraProvider, 
+  Box, 
+  Flex, 
+  Heading, 
+  Text, 
+  Button, 
+  Input, 
+  useColorMode, 
+  ColorModeScript, 
+  Link, 
+  InputGroup, 
+  InputRightElement,
+  FormControl 
+} from "@chakra-ui/react";
 
 function Login() {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [showPassword, setShowPassword] = useState(false);
+  const [isEmployer, setIsEmployer] = useState(false);
+  const { colorMode, toggleColorMode } = useColorMode();
+  const navigate = useNavigate();
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    if (!email || !password) {
+        alert("Please fill out login information");
+        return;
+    }
+
+    const response = await fetch("", {
+        method: "POST",
+        headers: {
+            "Content-Type": "application/json"
+        },
+        body: JSON.stringify({ email, password })
+    });
+
+    if (!response.ok)
+        throw new Error("Sign in request failed", response);
+
+    const data = await response.json();
+    console.log("User logged in successfully:", data);
+
+    // navigate("/search"); <--- uncomment this and comment rest of code to test routing
+
+    //checks if user is employer or job seeker
+    if(data.role === "employer") {
+      navigate("/employer-main");
+      setIsEmployer(true);
+    } else {
+      navigate("/search");
+      setIsEmployer(false);
+    }
+
+    return data;
+}
+
   return (
-    <div>
-      Login
-    </div>
+    <ChakraProvider>
+      <ColorModeScript initialColorMode="dark" />
+      <Box
+        p={[2, 4, 6, 8]}
+        maxWidth="100vw"
+        minHeight="100vh"
+        margin="auto"
+        bgGradient={colorMode === "light" ? 'linear(to-l, #663399, #D5B4F2)' : '#0B1215'}
+        color={colorMode === "light" ? "#000000" : "#F3F3F3"}
+        display="flex"
+        justifyContent="center"
+        alignItems="center"
+      >
+        <Flex direction="column" alignItems="center">
+          {/* idk how to make login box responsive */}
+          <Box 
+            bg={colorMode === "light" ? "#FFFFFF" : "#0B1215"} 
+            p={10} 
+            borderRadius="md" 
+            width="35vw" 
+            minHeight="65vh"
+          >
+            <Flex justifyContent="flex-end">
+
+              {/* change light/dark mode button layout to match w/ register */}
+              <Button
+                onClick={toggleColorMode}
+                mr={2}
+                color={colorMode === "light" ? "#FFFFFF" : "#000000"}
+                backgroundColor={colorMode === "light" ? "#000000" : "#A96CDE"}
```

Comment:
These sorts of checks for dark mode happen in quite a few places, it would probably be better if you made some sort of utils file/folder and moved the logic for these comparisons into some functions there